src/plat/pc99/machine/intel-vtd.c

```diff
@@ -265,6 +278,33 @@ void vtd_handle_fault(void)
     }
 }
 
+void vtd_create_irte(word_t interrupt_index, word_t vector, word_t pci_bus, word_t pci_dev, word_t pci_func)
+{
+    printf("IOMMU: adding interrupt remapping table entry %lu for vector %lu\n", interrupt_index, vector);
+
+    /* Remapping hardware units are configured to share the interrupt-remapping table,
+     * so we just read the table address from the first unit
+     */
+    if (x86KSnumDrhu == 0) {
+        return;
+    }
+    vtd_irte_t *irt = (vtd_irte_t *) paddr_to_pptr(vtd_read64(0, IRTA_REG) & ~0xfff);
+    vtd_irte_t *vtd_irte_slot = irt + interrupt_index;
```

Comment:
instead of explicit pointer arithmetic, can't we use `irt[interrupt_index] = vtd_irte_new(...)' ?